docs_src/design/adr/013-Device-Service-Events-Message-Bus.md

```diff
@@ -0,0 +1,215 @@
+
+The Message Bus abstraction allows for multiple subscriptions, so an Application Service could specify to receive data from only specific devices by creating multiple subscriptions. i.e.  `edgex/Events/Random-Integer-Device` and  `edgex/Events/Random-Boolean-Device`. Currently the App SDK only allows for a single subscription topic to be configured, but that could easily be expanded to handle a list of subscriptions. See [Configuration](#configuration) section below for details. 
+
+Core Data's existing publishing of Events would also need to be changed to use this new topic schema. One challenge with this is Core Data doesn't currently know the `Device Name` when it receives a CBOR encoded event. This is because it doesn't decode the Event until after it publishes it to the Message Bus. The V2 API could be enhanced to require the `Device Name` in the HTTP header when content type is CBOR.
+
+### Configuration
+
+#### Device Services
+
+All Device services will have the following additional configuration to allow connecting and publishing to the Message Bus.
+
+##### [MessageQueue]
+
+A  MessageQueue section will be added, which is similar to that used in Core Data today. To enable secure connections, the `Username` & `Password` have been replaced with `Authmode` & `SecretPath`, See **[Secure Connections](#secure-connections)** section below for details. The added `Enabled` property controls whether the Device Service publishes to the Message Bus or POSTs to Core Data. 
+
+```toml
+[MessageQueue]
+Enabled = true
+Protocol = 'tcp'
+Host = 'localhost'
+Port = 1883
+Type = 'mqtt'
+PublishTopic = 'edgex/events/<device-name>'
+[MessageQueue.Optional]
+    # Default MQTT Specific options that need to be here to enable evnironment variable overrides of them
+    # Client Identifiers
+    ClientId ="<device service key>"
+    # Connection information
+    Qos          =  "0" # Quality of Sevice values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+    KeepAlive    =  "10" # Seconds (must be 2 or greater)
+    Retained     = "false"
+    AutoReconnect  = "true"
+    ConnectTimeout = "5" # Seconds
+    # TLS configuration - Only used if Cert/Key file or Cert/Key PEMblock are specified
+    SkipCertVerify = "false"
+    Authmode = "none"
+    Secretpath = "messagebus"
+```
+
+The `PublishTopic` has a placeholder for the `Device Name` which gets replaced with the actual `Device Name`. If the place holder doesn't exist in the configured value, the  `PublishTopic` values  is used as is.
```

Comment:
I think this answers my question above.

Reply:
What if a user wanted different quality of service parameters or different secrets for each topic?

Reply:
As stated above there is only one connection to the message bus. These apply to that connection, not per topic.